crates/rspack_plugin_javascript/src/utils/eval/eval_binary_expr.rs

```diff
@@ -117,6 +117,87 @@ fn handle_abstract_equality_comparison(
   }
 }
 
+fn handle_logical_or(
```

Comment:
I believe this PR duplicates the content in `﻿rspack_plugin_javascript/src/utils/const/logic_expr`

Reply:
Actually they have some differences, such as the following example:

```js
// source code, false && true = false
if (typeof require !== 'function' && typeof require === 'function') { 
  require('fail');
}

// webpack, will remove dead code
if (false) {}

// rspack before, will calcualte each side but not a whole result, then dead code will be parsed
if ( false && 0) __webpack_require__(Object(function webpackMissingModule() { var e = new Error("Cannot find module 'fail'"); e.code = 'MODULE_NOT_FOUND'; throw e; }()));

// rspack after, will remove just like webpack
if (false) {}
```

Reply:
Oh, I see. There won't be any insertion of `﻿ConstDependency`